### ISSUE TITLE
UI scaling

### ISSUE THREAD

**Issue by Pimentoso** [maculateConception/aural-player#10]
This is more of a feature request than a bug. I like this player a lot but it's almost unusable on my work 27 inch imac, the font and the icons are just too small. See screenshot

https://imgur.com/a/DoK5M7s

Thanks for the hard work on this project.

**Comment by maculateConception** [comment]
Thanks a lot for the feedback and issue report.

One of my big limitations is that I only have 2 Mac machines to test my app - both are 13" Macbook Pros, with the max scaled resolution being 1680 x 1050.

To be honest, I'm not sure if I will be able to solve the issue you reported, simply because I do not have access to a machine/display with a larger resolution. The other challenge is - I'm not proficient enough with UI development to know how to scale a UI to work optimally across a wide range of display resolutions.

It is not surprising that this issue has finally come up after all this time. I did expect it, and again, I appreciate your report.

Despite not being confident that I will be able to resolve this issue, I will keep this issue open. Perhaps, someone from the community will know how to resolve this.

Cheers !
- maculateConception

**Comment by Pimentoso** [comment]
Yeah I have zero experience with UI building and the cocoa framework so I have no idea how much the scaling functions actually automagically work (usually they don't) and how big would this fix be. Thanks for taking the time to respond anyway. Please let me know if you need more screenshots, I can provide one for each scaling level.

**Comment by maculateConception** [comment]
Yes, I would love to see more screenshots showing things like the track name displayed (in the player window when a track is playing), and what would also be really useful is a comparison between Aural Player text and the text of another app (side by side) which shows text in a good font size for your display, so that I have a reference as to what would be a good size for your display.

**Comment by Pimentoso** [comment]
This is interesting, I tried it alongside the other program I'm using, and turns out it's not a matter of UI scaling, but only font size and icon spacing. The two programs scale identically when changing resolution.

https://imgur.com/a/j1Cn2AA

The main issues I see are

- font is too thin (and maybe a bit smaller)
- playlist items need more spacing
- control icons need more spacing too
- some icons are too cluttered (bottom playlist controls, volume/balance sliders)

So basically in the end it's not an issue but a matter of my personal preference. I think it would benefit just by simply spacing things a little.

**Comment by maculateConception** [comment]
@Pimentoso 

Hi, I am working on a fix for the issue you mentioned ... at least to fix the font size.

My new work should allow the user to choose between "Normal", "Larger", and "Largest" text font sizes.

As of now, I have done this only for player text resizing, as a trial. I plan to also allow resizing of Playlist and Effects window text.

See the GIF below:

<img width="1086" alt="Text sizes demo GIF" src="https://raw.githubusercontent.com/maculateConception/aural-player/master/Documentation/Demos/textSizes.gif">

**Comment by maculateConception** [comment]
@Pimentoso 

Hi, after a lot of thought and trial and error, I have implemented and released a solution to the text scaling problem. I now offer users 3 preset text size settings, that can be adjusted globally or per app window (see demo screenshot titled "Adjusting text size ..." on main page).

The new release is [here](https://github.com/maculateConception/aural-player/releases/tag/1.4.0)

If not ideal, I hope this will at least mitigate the issue you brought up, somewhat. Icon scaling/spacing is perhaps a bit more challenging, at least because of the constraints I have set up. I'll leave that for another day.

If you're able to try this out and let me know what you think, I'd really appreciate it.

Thanks again for bringing this to my attention. This issue is now closed.
